errno/errname.go

```diff
@@ -998,16 +998,16 @@ var MySQLErrName = map[uint16]*mysql.ErrMessage{
 	ErrWriteOnSnapshot:               mysql.Message("write on snapshot", nil),
 	ErrInvalidKey:                    mysql.Message("invalid key", nil),
 	ErrInvalidIndexKey:               mysql.Message("invalid index key", nil),
-	ErrDataInConsistent:              mysql.Message("index:%#v != record:%#v", []int{0, 1}),
+	ErrDataInconsistent:              mysql.Message("data inconsistency in table: %s, index: %s, handle: %s, index-values:%#v != record-values:%#v", []int{2, 3, 4}),
 	ErrDDLReorgElementNotExist:       mysql.Message("DDL reorg element does not exist", nil),
 	ErrDDLJobNotFound:                mysql.Message("DDL Job:%v not found", nil),
 	ErrCancelFinishedDDLJob:          mysql.Message("This job:%v is finished, so can't be cancelled", nil),
 	ErrCannotCancelDDLJob:            mysql.Message("This job:%v is almost finished, can't be cancelled now", nil),
 	ErrUnknownAllocatorType:          mysql.Message("Invalid allocator type", nil),
 	ErrAutoRandReadFailed:            mysql.Message("Failed to read auto-random value from storage engine", nil),
 	ErrInvalidIncrementAndOffset:     mysql.Message("Invalid auto_increment settings: auto_increment_increment: %d, auto_increment_offset: %d, both of them must be in range [1..65535]", nil),
-	ErrDataInConsistentExtraIndex:    mysql.Message("handle %#v, index:%#v != record:%#v", []int{0, 1, 2}),
-	ErrDataInConsistentMisMatchIndex: mysql.Message("col %s, handle %#v, index:%#v != record:%#v, compare err:%#v", []int{1, 2, 3, 4}),
+	ErrDataInconsistentMismatchCount: mysql.Message("data inconsistency in table: %s, index: %s, index-count:%d != record-count:%d", nil),
```

Comment:
for indexlookup detect, we only can find no match row problem(has index but has no matched row), but we can't know has row but no match index problem.

so for "no match row problem" we only can know miss row (not a != b)

but for performance reason, in table side of indexlookup, we only know the handles instead of index-rows info, so it's hard to give miss index row info

"admin check table" use a special way to maintain handle->index-row mapping to give miss index row info but it's not suitable to do that in normal indexlookup